Clean up docs to with regards to changes in SELFDESTRUCT opcode (#14803)
ethereum/solidity: docs/examples/micropayment.rst

an example of this in the first two lines of the ``claimPayment()``
function of the full contract at the end of this section.

Furthermore, instead of destroying the contract by calling ``selfdestruct``,

> which is currently deprecated, we will instead disable the contract's functionalities by freezing it,
> resulting in the reverting of any call after being frozen.

Reviewer: ```suggestion
which is currently deprecated, we will disable the contract's functionalities by freezing it,
resulting in the reversion of any call after it being frozen.
```

    // SPDX-License-Identifier: GPL-3.0
    pragma solidity >=0.7.0 <0.9.0;

    contract Owned {
        address payable owner;
        constructor() {
            owner = payable(msg.sender);
        }
    }

>     contract Frozeable is Owned {

Reviewer: ```suggestion
    contract Freezable is Owned {
```

        bool private _frozen = false;

        modifier notFrozen() {
            require(!_frozen, "Inactive Contract.");
            _;
        }

        function freeze() internal {
            if (msg.sender == owner)
                _frozen = true;
        }
    }

>     contract ReceiverPays is Frozeable {

Reviewer: ```suggestion
    contract ReceiverPays is Freezable {
```

When Bob is ready to receive his funds, it is time to
close the payment channel by calling a ``close`` function on the smart contract.
Closing the channel pays the recipient the Ether they are owed and

> deactivate the contract by freezing it, sending any remaining Ether back to Alice. To

Reviewer: ```suggestion
deactivates the contract by freezing it, sending any remaining Ether back to Alice. To
```
